src/LRUCache.ts

```diff
@@ -0,0 +1,53 @@
+export interface ILRUCache<K, V> {
+  get(key: K): V | undefined;
+  set(key: K, value: V): void;
+  clear(): void;
+}
+
+export class LRUCache<K, V> implements ILRUCache<K, V> {
```

Comment:
I would prefer to use an already tested and complete [package from NPM for LRU Cache](https://www.npmjs.com/search?q=lru-cache) rather than creating one from scratch.